extmod/modtime.c

```diff
@@ -100,10 +101,26 @@ MP_DEFINE_CONST_FUN_OBJ_1(mp_time_mktime_obj, time_mktime);
 
 #if MICROPY_PY_TIME_TIME_TIME_NS
 
+#if MICROPY_PY_TIME_TIME_FLOAT && !MICROPY_PY_BUILTINS_FLOAT
+#error "MICROPY_PY_TIME_TIME_FLOAT: Port does not have floating point support."
+#endif
+
+#if MICROPY_PY_TIME_TIME_FLOAT && !(MICROPY_FLOAT_IMPL >= MICROPY_FLOAT_IMPL_DOUBLE) && MICROPY_PY_TIME_TIME_FLOAT_WARNIMPRECISE
+#warning "MICROPY_PY_TIME_TIME_FLOAT: Port does not have adequate float precision. (`#define MICROPY_PY_TIME_TIME_FLOAT_WARNIMPRECISE (0)` to suppress.)"
+#endif
+
+#if MICROPY_PY_TIME_TIME_FLOAT && !MICROPY_PY_TIME_TIME_HAS_SUBSECOND && MICROPY_PY_TIME_TIME_FLOAT_WARNFUTILE
+#warning "MICROPY_PY_TIME_TIME_FLOAT: Port does not have sub-second information. (`#define MICROPY_PY_TIME_TIME_FLOAT_WARNFUTILE (0)` to suppress.)"
+#endif
```

Comment:
> Please no runtime warnings at functions that are subject to frequent use. Just leave that to the documentation. Its not only ports, but also boards of a port.

I only mean a set of compile-time `#warning`s like this. (And one `#error`, to give a slightly more useful message about a configuration that could never be made to build but likely for more opaque reasons otherwise.)